src/graph_notebook/magics/metadata.py

```diff
@@ -0,0 +1,141 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+"""
+
+import json
+import sys
+import re
+from typing import List
+from requests import Response
+from graph_notebook.visualization.template_retriever import retrieve_template
+pre_container_template = retrieve_template("pre_container.html")
+
+
+# class for individual metadata metric
+class Metric(object):
+    def __init__(self, name: str, friendly_name: str, default_value: any = "N/A"):
+        self.name = name    # used as key metadata.metrics
+        self.friendly_name = friendly_name
+        self.value = default_value
+
+    def set_value(self, value: any):
+        self.value = value
+
+
+# class for complete metadata object to output
+class Metadata(object):
+    def __init__(self):
+        self.metrics = {}
+
+    def insert_metric(self, new_metric: Metric):
+        self.metrics[new_metric.name] = new_metric
+
+    def bulk_insert_metrics(self, new_metrics: List[Metric]):
+        for metric in new_metrics:
+            self.insert_metric(metric)
+
+    def set_metric_value(self, metric_name, value):
+        self.metrics[metric_name].set_value(value)
+
+    def set_request_metrics(self, res: Response):
+        self.set_metric_value('request_time', 1000 * res.elapsed.total_seconds())
+        self.set_metric_value('status', res.status_code)
+        self.set_metric_value('status_ok', res.ok)
+        self.set_metric_value('resp_size', sys.getsizeof(res.content))
+
+    def set_gremlin_profile_metrics(self, profile: str):
```

Comment:
This probably doesn't fit in the Metadata class itself. Instead, it could accept a Metadata object and a gremlin profile string and add the appropriate metrics to the metadata.

Reply:
Done